lib/timers.js

```diff
@@ -500,8 +500,30 @@ function rearm(timer) {
   }
 }
 
+function extractTimer(timer) {
+  if (Timeout.prototype.isPrototypeOf(timer) || typeof timer !== 'object' ||
+      timer === null || timer === undefined) {
```

Comment:
There is a case that need considering:
https://github.com/nodejs/node/blob/472a66517a2a34fdce77cd0f6f29a2e62e7fd5d9/lib/net.js#L381-L396
(this is something we want to refactor in the future Ref: https://github.com/nodejs/node/issues/14209)